dna/modify.go

```diff
@@ -133,6 +139,23 @@ func Insert(seq []Base, insPos int64, insSeq []Base) []Base {
 	return append(seq[:insPos], append(insSeq, seq[insPos:]...)...)
 }
 
+// InsertStable is identical to Insert, but conserves the
+// underlying memory of the input destSeq slice if possible
+func InsertStable(destSeq []Base, insPos int, insSeq []Base) ([]Base, error) {
+	if insPos < 0 || insPos > len(destSeq) {
+		return nil, ErrInvalidPosition
+	}
+	if len(destSeq)+len(insSeq) > cap(destSeq) {
+		return Insert(destSeq, int64(insPos), insSeq), ErrExceedCapNewSliceCreated
+	}
+	destSeq = destSeq[:len(destSeq)+1]                                           // extend the LEN of the slice to include the inserted bases
+	copy(destSeq[insPos+len(insSeq):cap(destSeq)], destSeq[insPos:cap(destSeq)]) // shift bases in slice by len(insSeq) at the position of the insertion
```

Comment:
I would consider using len instead of cap.
copy(destSeq[insPos+len(insSeq):len(destSeq)+len(insSeq)], destSeq[insPos:len(destSeq)])
I think the way you have it now you will be potentially copying a lot of values that are not in the length of destSeq, but are in the capacity.  For example, picture destSeq having a huge capacity (1 billion bases), but only using 1kb right now.  You will still be copying all the memory values between 1kb and 1billion.